### ISSUE TITLE
Compatibility with Server 10.5

### ISSUE THREAD

**Issue by pmaier1** [owncloud/richdocuments#346]
- [x] Check supported PHP versions (7.2, 7.3, 7.4)
- [ ] Use the new licensing mechanism. #347 
  - [ ] Run a manual test with secure view, grace period and demo key @pmaier who has an instance
- [x] Check whether there's anything else to do for 10.5 compatibility (e.g., `min-version = 10.5`)
- [ ] Create release ticket and do the release

**Comment by cdamken** [comment]
## LOGS
<details><summary>CLICK ME</summary>
<p>

```
{
  "reqId": "OYQM7JDtqEa2QgDjpX8g",
  "level": 1,
  "time": "2020-07-18T00:54:47+02:00",
  "remoteAddr": "189.216.180.55",
  "user": "aries-server",
  "app": "richdocuments",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/index?fileId=16073918&dir=%2FCONCURSOS%202018%2FCAEM%2FSOCIOECONOMICO",
  "message": "getWopiInfoForAuthUser(): Generating WOPI Token for file 16073918, version 0."
}
{
  "reqId": "3iwipN6VXsAPJJColEsp",
  "level": 1,
  "time": "2020-07-18T00:54:51+02:00",
  "remoteAddr": "127.0.0.1",
  "user": "--",
  "app": "richdocuments",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/wopi/files/16073918_ocuhsyihz2sp_0_0?access_token=IvsFFAQvwPElwV6TfYPq4n8CYkc8w28u&access_token_ttl=0&permission=edit",
  "message": "wopiCheckFileInfo(): Getting info about file 16073918, version 0 by token IvsFFAQvwPElwV6TfYPq4n8CYkc8w28u."
}
{
  "reqId": "lgBCmu6mVPV9JBEgLpb9",
  "level": 1,
  "time": "2020-07-18T00:54:51+02:00",
  "remoteAddr": "127.0.0.1",
  "user": "--",
  "app": "richdocuments",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/wopi/files/16073918_ocuhsyihz2sp_0_0/contents?access_token=IvsFFAQvwPElwV6TfYPq4n8CYkc8w28u&access_token_ttl=0&permission=edit",
  "message": "wopiGetFile(): File 16073918, version 0, token IvsFFAQvwPElwV6TfYPq4n8CYkc8w28u."
}
{
  "reqId": "XYVNzKeCvtywqJaBnsHO",
  "level": 1,
  "time": "2020-07-18T00:59:26+02:00",
  "remoteAddr": "189.216.180.55",
  "user": "aries-server",
  "app": "richdocuments",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/index?fileId=16073781&dir=%2FCONCURSOS%202018%2FCAEM%2FESTUDIOS%20Y%20PROYECTOS",
  "message": "getWopiInfoForAuthUser(): Generating WOPI Token for file 16073781, version 0."
}
{
  "reqId": "dsjYrzVNiVwX6vC6hrMo",
  "level": 1,
  "time": "2020-07-18T00:59:30+02:00",
  "remoteAddr": "127.0.0.1",
  "user": "--",
  "app": "richdocuments",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/wopi/files/16073781_ocuhsyihz2sp_0_0?access_token=f2jdIMrI4KPSQHIW2QH2hThKk6oXlJM4&access_token_ttl=0&permission=edit",
  "message": "wopiCheckFileInfo(): Getting info about file 16073781, version 0 by token f2jdIMrI4KPSQHIW2QH2hThKk6oXlJM4."
}
{
  "reqId": "xsspbR1mRwTTl8ukKiA3",
  "level": 1,
  "time": "2020-07-18T00:59:30+02:00",
  "remoteAddr": "127.0.0.1",
  "user": "--",
  "app": "richdocuments",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/wopi/files/16073781_ocuhsyihz2sp_0_0/contents?access_token=f2jdIMrI4KPSQHIW2QH2hThKk6oXlJM4&access_token_ttl=0&permission=edit",
  "message": "wopiGetFile(): File 16073781, version 0, token f2jdIMrI4KPSQHIW2QH2hThKk6oXlJM4."
}
{
  "reqId": "w4MgzHKyrhOwBvGaArDu",
  "level": 1,
  "time": "2020-07-18T01:00:19+02:00",
  "remoteAddr": "189.216.180.55",
  "user": "aries-server",
  "app": "richdocuments",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/index?fileId=16073583&dir=%2FCONCURSOS%202018%2FCAEM%2FESTUDIOS%20Y%20PROYECTOS%2FOFICIOS",
  "message": "getWopiInfoForAuthUser(): Generating WOPI Token for file 16073583, version 0."
}
{
  "reqId": "4C2ODan0NIBg5xT8HL3A",
  "level": 1,
  "time": "2020-07-18T01:00:23+02:00",
  "remoteAddr": "127.0.0.1",
  "user": "--",
  "app": "richdocuments",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/wopi/files/16073583_ocuhsyihz2sp_0_0?access_token=YbEvJIEu3KHoEuxKE6Ee0KvpSK6HP8Ox&access_token_ttl=0&permission=edit",
  "message": "wopiCheckFileInfo(): Getting info about file 16073583, version 0 by token YbEvJIEu3KHoEuxKE6Ee0KvpSK6HP8Ox."
}
{
  "reqId": "2yUDR8Z5M4UIlpACIo8A",
  "level": 1,
  "time": "2020-07-18T01:00:23+02:00",
  "remoteAddr": "127.0.0.1",
  "user": "--",
  "app": "richdocuments",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/wopi/files/16073583_ocuhsyihz2sp_0_0/contents?access_token=YbEvJIEu3KHoEuxKE6Ee0KvpSK6HP8Ox&access_token_ttl=0&permission=edit",
  "message": "wopiGetFile(): File 16073583, version 0, token YbEvJIEu3KHoEuxKE6Ee0KvpSK6HP8Ox."
}
{
  "reqId": "sByxVARsapMVJ8TkpcF4",
  "level": 1,
  "time": "2020-07-19T20:51:50+02:00",
  "remoteAddr": "85.212.99.98",
  "user": "eneubauer",
  "app": "richdocuments",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/index?fileId=16359542&dir=%2Fdocs",
  "message": "getWopiInfoForAuthUser(): Generating WOPI Token for file 16359542, version 0."
}
{
  "reqId": "AFhf9dlD8Rlj06KY3mtp",
  "level": 1,
  "time": "2020-07-19T20:51:56+02:00",
  "remoteAddr": "127.0.0.1",
  "user": "--",
  "app": "richdocuments",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/wopi/files/16359542_ocuhsyihz2sp_0_0?access_token=rsCnhSSdQVp27jbKMqNXG7av7RDbmOMb&access_token_ttl=0&permission=edit",
  "message": "wopiCheckFileInfo(): Getting info about file 16359542, version 0 by token rsCnhSSdQVp27jbKMqNXG7av7RDbmOMb."
}
{
  "reqId": "BfHBmI5gBhxSkaF7znA9",
  "level": 1,
  "time": "2020-07-19T20:51:56+02:00",
  "remoteAddr": "127.0.0.1",
  "user": "--",
  "app": "richdocuments",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/wopi/files/16359542_ocuhsyihz2sp_0_0/contents?access_token=rsCnhSSdQVp27jbKMqNXG7av7RDbmOMb&access_token_ttl=0&permission=edit",
  "message": "wopiGetFile(): File 16359542, version 0, token rsCnhSSdQVp27jbKMqNXG7av7RDbmOMb."
}
{
  "reqId": "z000tkWIwBEdBEQPzjd7",
  "level": 2,
  "time": "2020-07-19T20:53:04+02:00",
  "remoteAddr": "127.0.0.1",
  "user": "eneubauer",
  "app": "richdocuments",
  "method": "POST",
  "url": "/index.php/apps/richdocuments/wopi/files/16359542_ocuhsyihz2sp_0_0/contents?access_token=rsCnhSSdQVp27jbKMqNXG7av7RDbmOMb&access_token_ttl=0&permission=edit",
  "message": "wopiPutFile(): mtime"
}
{
  "reqId": "f1yG3gJ4L4iRrRLAzLRu",
  "level": 1,
  "time": "2020-07-19T20:55:31+02:00",
  "remoteAddr": "85.212.99.98",
  "user": "eneubauer",
  "app": "richdocuments",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/index?fileId=16359542&dir=%2Fdocs",
  "message": "getWopiInfoForAuthUser(): Generating WOPI Token for file 16359542, version 0."
}
{
  "reqId": "98DVaVxdRegibksItHTT",
  "level": 1,
  "time": "2020-07-19T20:55:35+02:00",
  "remoteAddr": "127.0.0.1",
  "user": "--",
  "app": "richdocuments",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/wopi/files/16359542_ocuhsyihz2sp_0_0?access_token=VUX3YZbVAXPotv63viaCsyXnRvr9ltg5&access_token_ttl=0&permission=edit",
  "message": "wopiCheckFileInfo(): Getting info about file 16359542, version 0 by token VUX3YZbVAXPotv63viaCsyXnRvr9ltg5."
}
{
  "reqId": "rVHdiBgnpLB2pMLtsdD7",
  "level": 1,
  "time": "2020-07-19T20:55:35+02:00",
  "remoteAddr": "127.0.0.1",
  "user": "--",
  "app": "richdocuments",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/wopi/files/16359542_ocuhsyihz2sp_0_0/contents?access_token=VUX3YZbVAXPotv63viaCsyXnRvr9ltg5&access_token_ttl=0&permission=edit",
  "message": "wopiGetFile(): File 16359542, version 0, token VUX3YZbVAXPotv63viaCsyXnRvr9ltg5."
}
{
  "reqId": "PAN6zFUlv5q6ieAQI3hO",
  "level": 1,
  "time": "2020-07-19T20:55:38+02:00",
  "remoteAddr": "85.212.99.98",
  "user": "eneubauer",
  "app": "richdocuments",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/index?fileId=16359542&dir=%2Fdocs",
  "message": "getWopiInfoForAuthUser(): Generating WOPI Token for file 16359542, version 0."
}
{
  "reqId": "wHCoMBGe4biI6EjeHhr4",
  "level": 1,
  "time": "2020-07-19T20:55:40+02:00",
  "remoteAddr": "127.0.0.1",
  "user": "--",
  "app": "richdocuments",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/wopi/files/16359542_ocuhsyihz2sp_0_0?access_token=Foec4oW2ySou1EWZi1TufO3Gwy8ygv99&access_token_ttl=0&permission=edit",
  "message": "wopiCheckFileInfo(): Getting info about file 16359542, version 0 by token Foec4oW2ySou1EWZi1TufO3Gwy8ygv99."
}
{
  "reqId": "n7b1Ytk3SrWiQlwAXzGU",
  "level": 3,
  "time": "2020-07-19T20:56:52+02:00",
  "remoteAddr": "85.212.99.98",
  "user": "eneubauer",
  "app": "PHP",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/index?fileid=16359542",
  "message": "Undefined index: version at /var/www/owncloud/apps/richdocuments/templates/documents.php#3"
}
{
  "reqId": "n7b1Ytk3SrWiQlwAXzGU",
  "level": 3,
  "time": "2020-07-19T20:56:52+02:00",
  "remoteAddr": "85.212.99.98",
  "user": "eneubauer",
  "app": "PHP",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/index?fileid=16359542",
  "message": "Undefined index: sessionId at /var/www/owncloud/apps/richdocuments/templates/documents.php#4"
}
{
  "reqId": "n7b1Ytk3SrWiQlwAXzGU",
  "level": 3,
  "time": "2020-07-19T20:56:52+02:00",
  "remoteAddr": "85.212.99.98",
  "user": "eneubauer",
  "app": "PHP",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/index?fileid=16359542",
  "message": "Undefined index: permissions at /var/www/owncloud/apps/richdocuments/templates/documents.php#5"
}
{
  "reqId": "n7b1Ytk3SrWiQlwAXzGU",
  "level": 3,
  "time": "2020-07-19T20:56:52+02:00",
  "remoteAddr": "85.212.99.98",
  "user": "eneubauer",
  "app": "PHP",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/index?fileid=16359542",
  "message": "Undefined index: title at /var/www/owncloud/apps/richdocuments/templates/documents.php#6"
}
{
  "reqId": "n7b1Ytk3SrWiQlwAXzGU",
  "level": 3,
  "time": "2020-07-19T20:56:52+02:00",
  "remoteAddr": "85.212.99.98",
  "user": "eneubauer",
  "app": "PHP",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/index?fileid=16359542",
  "message": "Undefined index: fileId at /var/www/owncloud/apps/richdocuments/templates/documents.php#7"
}
{
  "reqId": "n7b1Ytk3SrWiQlwAXzGU",
  "level": 3,
  "time": "2020-07-19T20:56:52+02:00",
  "remoteAddr": "85.212.99.98",
  "user": "eneubauer",
  "app": "PHP",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/index?fileid=16359542",
  "message": "Undefined index: token at /var/www/owncloud/apps/richdocuments/templates/documents.php#8"
}
{
  "reqId": "n7b1Ytk3SrWiQlwAXzGU",
  "level": 3,
  "time": "2020-07-19T20:56:52+02:00",
  "remoteAddr": "85.212.99.98",
  "user": "eneubauer",
  "app": "PHP",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/index?fileid=16359542",
  "message": "Undefined index: urlsrc at /var/www/owncloud/apps/richdocuments/templates/documents.php#9"
}
{
  "reqId": "n7b1Ytk3SrWiQlwAXzGU",
  "level": 3,
  "time": "2020-07-19T20:56:53+02:00",
  "remoteAddr": "85.212.99.98",
  "user": "eneubauer",
  "app": "PHP",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/index?fileid=16359542",
  "message": "Undefined index: path at /var/www/owncloud/apps/richdocuments/templates/documents.php#10"
}
{
  "reqId": "n7b1Ytk3SrWiQlwAXzGU",
  "level": 3,
  "time": "2020-07-19T20:56:53+02:00",
  "remoteAddr": "85.212.99.98",
  "user": "eneubauer",
  "app": "PHP",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/index?fileid=16359542",
  "message": "Undefined index: uploadMaxHumanFilesize at /var/www/owncloud/apps/richdocuments/templates/documents.php#32"
}
{
  "reqId": "n7b1Ytk3SrWiQlwAXzGU",
  "level": 3,
  "time": "2020-07-19T20:56:53+02:00",
  "remoteAddr": "85.212.99.98",
  "user": "eneubauer",
  "app": "PHP",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/index?fileid=16359542",
  "message": "Undefined index: uploadMaxFilesize at /var/www/owncloud/apps/richdocuments/templates/documents.php#40"
}
{
  "reqId": "n7b1Ytk3SrWiQlwAXzGU",
  "level": 3,
  "time": "2020-07-19T20:56:53+02:00",
  "remoteAddr": "85.212.99.98",
  "user": "eneubauer",
  "app": "PHP",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/index?fileid=16359542",
  "message": "Undefined index: uploadMaxFilesize at /var/www/owncloud/apps/richdocuments/templates/documents.php#42"
}
{
  "reqId": "n7b1Ytk3SrWiQlwAXzGU",
  "level": 3,
  "time": "2020-07-19T20:56:53+02:00",
  "remoteAddr": "85.212.99.98",
  "user": "eneubauer",
  "app": "PHP",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/index?fileid=16359542",
  "message": "Undefined index: uploadMaxHumanFilesize at /var/www/owncloud/apps/richdocuments/templates/documents.php#48"
}
{
  "reqId": "p085zQpSNO5wD4G3TowT",
  "level": 1,
  "time": "2020-07-19T20:57:00+02:00",
  "remoteAddr": "85.212.99.98",
  "user": "eneubauer",
  "app": "richdocuments",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/index?fileId=16359542",
  "message": "getWopiInfoForAuthUser(): Generating WOPI Token for file 16359542, version 0."
}
{
  "reqId": "mpWUCgQj1soRR190hJId",
  "level": 1,
  "time": "2020-07-19T20:57:03+02:00",
  "remoteAddr": "127.0.0.1",
  "user": "--",
  "app": "richdocuments",
  "method": "GET",
  "url": "/index.php/apps/richdocuments/wopi/files/16359542_ocuhsyihz2sp_0_0?access_token=5IbJSl9nDrDwz9i2O450IEwd5lpAjMbF&access_token_ttl=0&permission=edit",
  "message": "wopiCheckFileInfo(): Getting info about file 16359542, version 0 by token 5IbJSl9nDrDwz9i2O450IEwd5lpAjMbF."
}

```
</p>
</details>

**Comment by mrow4a** [comment]
For richdocuments upgrade path is not that streightforward:
```
Exception: {\"Exception\":\"OCP\\\\AppFramework\\\\QueryException\",\"Message\":\"Could not resolve uid! Class uid does not exist or error resolving constructor arguments: Could not resolve LicenseManager! Class LicenseManager does not exist\
```

**Comment by mrow4a** [comment]
fixed in:
- https://github.com/owncloud/core/pull/37728
- https://github.com/owncloud/richdocuments/pull/347#issuecomment-663958514

**Comment by phil-davis** [comment]
#347 has been merged. (core code was merged some days ago)

**Comment by jnweiger** [comment]
We need to releease a fix here:
Enabling richdocuments should not immediately start the enterprise license grace period process. 
Only the secure view feature is an enterprise feature -- apart from that, it is a community app.

**Comment by pmaier1** [comment]
This has been fixed with the release of `richdocuments` 2.4.1 (https://marketplace.owncloud.com/apps/richdocuments). If you're stuck on Server 10.5 and `richdocuments` 2.4.0 (e.g., grace period active asking for trial key or `richdocuments` disabled and can't be enabled again), please do the following to continue using Collabora Online:

**Uninstall `richdocuments` 2.4.0 and install 2.4.1, e.g., using the Market app.**
